Update `frequenz-api-common` to v0.7.0

tiyash-basu-frequenz opened the pull request.

tiyash-basu-frequenz committed:
Update microgrid.proto to use ElectricalComponent terminology

The term `Component` has been changes in the common API to
`ElectricalComponent`. This commit introduces this new terminology in
the Microgrid API as well, replacing all instances of `Component` with
`ElectricalComponent`. This change is made to maintain consistency
across the API and to clarify that the components being referred to are
electrical in nature.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Remove sensor categories from `ListSensorsRequest`

Sensor categorization is no further suported in the common-API.
This commit reflects that change by removing the `categories` field
from the `ListSensorsRequest` message in the Microgrid API.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Update formatting

This commit updates the formatting of the `microgrid.proto` file in the
following ways:
- RPC method descriptions are split into multiple lines for better
  readability.
- Documentation is limited to 80 characters per line.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

stefan-brus-frequenz reviewed (dismissed):
LGTM

llucax commented on proto/frequenz/api/microgrid/v1/microgrid.proto:
```diff
@@ -296,89 +300,90 @@ service Microgrid {
+// Contains filtering parameters for listing electrical components.
+message ListElectricalComponentsRequest {
+  // Return electrical components that have the specified IDs only.
+  repeated uint64 electrical_component_ids = 1;
```
Since this got so long now, I wonder if we really want to have the extra verbosity in request/responses, or maybe be more succinct and use just `ids`, `categories` and `components` in the fields instead of the full `electrical_component_` prefix, as *electrical component* is already in the `Message` and `rpc` names.

tiyash-basu-frequenz replied:
I considered that, and decided against it. The challenge is we lose a lot of specificity this way. Given the message is named `ListElectricalComponentsRequest`, it will not be clear straightaway what `ids` might mean, and users will have to refer to the documentation. So while it would make the API less verbose, the developer experience might take a hit.

Hence the expanded versions.

tiyash-basu-frequenz replied:
FWIW, I get your point about the verbosity. I'll create an issue for this topic once these changes are in.

llucax commented on proto/frequenz/api/microgrid/v1/microgrid.proto:
```diff
@@ -296,89 +300,90 @@ service Microgrid {
   // The list of connections.
-  repeated frequenz.api.common.v1.microgrid.components.ComponentConnection
+  repeated frequenz.api.common.v1alpha7.microgrid.electrical_components.ElectricalComponentConnection
     connections = 1;
```
Whichever route we take, I think it would be good to be consistent, so if we are using prefixes, I would make this `electrical_component_connections`, but I think I prefer the short versions, it seems like at the stage of creating/reading a message it is already obvious that we are talking about electrical components from the message and RPC names.

tiyash-basu-frequenz replied:
I would opt for the longer version, for maintaining consistency with the other field names.

llucax replied:
OK, so rename this to `electrical_component_connections` then?

tiyash-basu-frequenz replied:
Done.

llucax commented on proto/frequenz/api/microgrid/v1/microgrid.proto:
```diff
@@ -388,28 +393,28 @@ message ReceiveComponentDataStreamRequest {
+// Electrical Components belonging to all categories and types can be
+// represented using this message.
+message ReceiveElectricalComponentDataStreamResponse {
+  frequenz.api.common.v1alpha7.microgrid.electrical_components.ElectricalComponentTelemetry data = 1;
```
Also wonder if we want to change this to `telemetry` to make it consistent with the message name.

tiyash-basu-frequenz replied:
done

llucax commented on RELEASE_NOTES.md:
```diff
@@ -9,6 +9,10 @@
 - The minimum supported version of `protobuf` was bumped to 5.28.1 and to 1.66.1 for `grpcio`. Make sure to update your dependencies accordingly.
 - The `AddComponentBoundsRequest.validity_duration` field was replaced by `AddComponentBoundsRequest.request_lifetime` to match the fields in `SetComponentPowerActiveRequest` and `SetComponentPowerReactiveRequest`. The default value was also changed from 5 to 60 seconds.
 - The `AddComponentBoundsResponse.ts` field was renamed to `AddComponentBoundsResponse.valid_until` for extra clarity and also to match the fields in `SetComponentPowerActiveResponse` and `SetComponentPowerReactiveResponse`.
+- The minimum required version of `frequenz-repo-config` was updated to 0.13.5.
```
I would remove this, as this is just a build/development dependency, downstream users don't even need to know this project uses repo-config.
```suggestion
```

tiyash-basu-frequenz replied:
done

llucax commented on pyproject.toml:
```diff
@@ -33,7 +33,7 @@ classifiers = [
 requires-python = ">= 3.11, < 4"
 dependencies = [
-  "frequenz-api-common >= 0.6.5, < 0.7",
+  "frequenz-api-common @ git+https://github.com/frequenz-floss/frequenz-api-common@v0.x.x",
```
I guess this sneaked into this commit accidentally?

tiyash-basu-frequenz replied:
removed

tiyash-basu-frequenz committed:
Update frequenz-api-common dependency to v0.7.0

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Renamed telemetry fields in Microgrid API to match common API

This commit renames the folowingfields:
- `ReceiveComponentDataStreamResponse.data` to
  `ReceiveElectricalComponentDataStreamResponse.telemetry`
- `ReceiveSensorDataStreamResponse.data` to
  `ReceiveSensorDataStreamResponse.telemetry`

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

llucax commented on RELEASE_NOTES.md:
```diff
@@ -9,6 +9,11 @@
 - The minimum supported version of `protobuf` was bumped to 5.28.1 and to 1.66.1 for `grpcio`. Make sure to update your dependencies accordingly.
 - The `AddComponentBoundsRequest.validity_duration` field was replaced by `AddComponentBoundsRequest.request_lifetime` to match the fields in `SetComponentPowerActiveRequest` and `SetComponentPowerReactiveRequest`. The default value was also changed from 5 to 60 seconds.
 - The `AddComponentBoundsResponse.ts` field was renamed to `AddComponentBoundsResponse.valid_until` for extra clarity and also to match the fields in `SetComponentPowerActiveResponse` and `SetComponentPowerReactiveResponse`.
+- The version of `frequenz-api-common` has been updated to v0.7.0. Please refer to the [release notes](https://github.com/frequenz-floss/frequenz-api-common/releases/tag/v0.7.0) and update your code accordingly.
```
Commits still seem a bit screwed. "Bump frequenz-repo-config to 0.13.5" is advancing the `frequenz-api-common` submodule instead of the commit bumping `frequenz-api-common` to v0.7.0.

tiyash-basu-frequenz replied:
![image](https://github.com/user-attachments/assets/04668ac1-d816-4c24-9e91-64f4841b8eb8)
Interesting. I do not see that, or maybe I am misunderstanding something. I have added a screenshot to share what I see. I have not changed these 2 commits since your comment here.

llucax replied:
Yeah, I mean the **git submodule**

![image](https://github.com/user-attachments/assets/3bfbceb1-bbac-4f17-96b8-e0ae185c5b2a)

tiyash-basu-frequenz replied:
I noticed that the submodule update was happening in the wrong commit. Fixed now.

tiyash-basu-frequenz commented:
I added another commit to rename start and end connection fields. As mentioned [here](https://github.com/frequenz-floss/frequenz-api-microgrid/pull/319#discussion_r2175145218), I'll address the verbosity in a different issue, where I can focus properly on it.

tiyash-basu-frequenz commented:
Rebased.

llucax commented on RELEASE_NOTES.md:
```diff
@@ -9,6 +9,20 @@
+  - `ListComponentConnectionsResponse.connections` has been renamed to `ListElectricalComponentConnectionsResponse.electrical_component_connections` to match the naming conventions used in `frequenz-api-common`.
+  - `ReceiveComponentDataStreamResponse.data` has been renamed to `ReceiveElectricalComponentDataStreamResponse.telemetry` to match the naming conventions used in `frequenz-api-common`.
+  - `ReceiveSensorDataStreamResponse.data` has been renamed to `ReceiveSensorDataStreamResponse.telemetry` to match the naming conventions used in `frequenz-api-common`.
+  - `ListComponentConnectionsRequest.starts` has been renamed to `ListElectricalComponentConnectionsRequest.from_electrical_component_id` to match the naming conventions used in `frequenz-api-common`.
+  - `ListComponentConnectionsRequest.ends` has been renamed to `ListElectricalComponentConnectionsRequest.to_electrical_component_id` to match the naming conventions used in `frequenz-api-common`.
```
Nitpick: What about `source`/`destination_electrical_component_id` as proposed in https://github.com/frequenz-floss/frequenz-api-microgrid/issues/262? The new version of the client is already using `source` and `destination` (or course I can change it too).

But when reading the "from/to" it felt like the meaning was more like "from electrical component to sensor" or something like that, it felt like "from" makes "electrical component" highlight more and expect a different "to" while with "source electrical component" makes my brain think then I should expect a destination of the same type (electrical component).

But also if you are talking about properties of a connection, you can talk about a connection source or destination, while a connection from or to sounds super weird.

tiyash-basu-frequenz replied:
The reason I avoided the `source`/`destination` terminology is because it might reinforce a wrong idea of positive energy flow direction. I have created issue #325 to address it independently of this PR. For now, I will remove this commit from this PR.

tiyash-basu-frequenz replied:
> For now, I will remove this commit from this PR.

Done

tiyash-basu-frequenz committed:
Rename RPC methods for streaming data

This commit renames the RPC methods for streaming data in the microgrid
API to match the naming conventions used in `frequenz-api-common`. The
changes include:
- `ReceiveElectricalComponentDataStream` has been renamed to
  `ReceiveElectricalComponentTelemetryStream`.
- `ReceiveElectricalComponentDataStreamRequest` has been renamed to
  `ReceiveElectricalComponentTelemetryStreamRequest`.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

llucax commented on pyproject.toml:
Now the update to repo config is completely missing here and the commit that says it is doing the bump only has a documentation fix :grimacing: 

![image](https://github.com/user-attachments/assets/ce08dd60-630c-4b83-bdaa-61e767ea5714)

tiyash-basu-frequenz replied:
>Now the update to repo config is completely missing here

Ah looks like one of the dependabot PRs upgraded repo config. I have removed the commit.
